test/distributed/rpc/test_share_memory.py

```diff
@@ -0,0 +1,82 @@
+import torch
```

Comment:
I based this test file on https://github.com/pytorch/pytorch/pull/66970#issuecomment-1063643284

Maybe there's a simpler way to test this, but I'm not sure how